vyper/codegen/stmt.py

```diff
@@ -255,7 +255,7 @@ def _parse_For_range(self):
             with end.cache_when_complex("end") as (b1, end):
                 # note: the check for rounds<=rounds_bound happens in asm
                 # generation for `repeat`.
-                clamped_start = clamp("le", start, end)
+                clamped_start = clamp("sle" if target_type.is_signed else "le", start, end)
```

Comment:
```suggestion
                le = "le"
                if target_type.is_signed:
                    le = "sle"
                clamped_start = clamp(LE, start, end)
```

Reply:
ah right, check `clamp2` in `vyper/codegen/core.py`

Reply:
https://github.com/vyperlang/vyper/pull/3814/commits/f8412ee90957cf738fcd5fc76842a666fafbe84d